src/main/java/org/terasology/simpleFarming/systems/TreeAuthoritySystem.java

```diff
@@ -0,0 +1,573 @@
+	 * @param log The log to be destroyed.
+	 * 
+	 * @see #destroyLeaves(Set, DoDestroyEvent)
+	 */
+	private void destroyLog(DoDestroyEvent event, EntityRef log) {
+		LogComponent logComponent = log.getComponent(LogComponent.class);
+		EntityRef rootEntity = logComponent.root;
+		RootComponent rootComponent = rootEntity.getComponent(RootComponent.class);
+		
+		if(rootComponent.alive) {
+			destroyLeaves(rootComponent.leaves, event);
+			rootComponent.leaves.clear();
+			rootComponent.alive = false;
+			rootEntity.addOrSaveComponent(rootComponent);
+		}
+		
+		if(event == null) {
+			worldProvider.setBlock(logComponent.location, airBlock);
+		}
+		
+		Vector3i aboveLocation = new Vector3i(logComponent.location).addY(1);
+		EntityRef aboveLogEntity = blockEntityRegistry.getExistingEntityAt(aboveLocation);
+		if(aboveLogEntity != EntityRef.NULL && aboveLogEntity.hasComponent(LogComponent.class)
+				&& aboveLogEntity.getComponent(LogComponent.class).root == rootEntity) {
+			if(event == null) {
+				destroyLog(null, aboveLogEntity);
+			} else {
+				aboveLogEntity.send(event);
+			}
+		}
+	}
+
+	/**
+	 * Destroys all of the leaves in a set of leaves.
+	 * 
+	 * @param leaves The set of leaves to destroy.
+	 * @param event The event for the leaves being destroyed. If it is present, it will be sent to all blocks which are destroyed,
+	 * allowing them to drop items, otherwise, the blocks will simply be replaced with air, allowing for growth or un-growth.
+	 * 
+	 * @see RootComponent#leaves
+	 */
+	private void destroyLeaves(Set<EntityRef> leaves, DoDestroyEvent event) {
+		for(EntityRef leaf : leaves) {
+			if(leaf.exists() && leaf.hasComponent(BushDefinitionComponent.class)) {
+				Vector3i leafLocation = leaf.getComponent(BlockComponent.class).position;
+				if(event != null) {
+					leaf.send(event);
+				} else {
+					worldProvider.setBlock(leafLocation, airBlock);
+				}
+			}
+		}
+	}
+
+	/**
+     * Starts a new growth timer with random duration, subject to the given bounds.
+     * If a growth timer had already been started, the previously scheduled event will be removed.
+     *
+     * @param entity the entity to set the timer on
+     * @param min    the minimum duration in milliseconds
+     * @param max    the maximum duration in milliseconds
+     */
+    private void resetDelay(EntityRef entity, int min, int max) {
+    	String actionId = "SimpleFarming:" + entity.getId();
+    	if(delayManager.hasDelayedAction(entity, actionId)) {
+    		delayManager.cancelDelayedAction(entity, actionId);
+    	}
+        delayManager.addDelayedAction(entity, actionId, generateRandom(min, max));
+    }
+
+    /**
+     * Returns a random integer in the specified interval.
+     *
+     * @param min The minimum number
+     * @param max The maximum number
+     * @return the random number, or {@code min} if {@code max <= min}
+     */
+    private long generateRandom(int min, int max) {
```

Comment:
So, this is nothing to do with your coding, but now we have three systems which are using the exact same method we could probably move it to a static method on something like the main `PlantAuthoritySystem`.